### PR TITLE
chore!: Remove manual serialization of `Opcode`s in favour of `serde`

### PR DESCRIPTION
**Pull request by kevaundray**
# Related issue(s)

(If it does not already exist, first create a GitHub issue that describes the problem this Pull Request (PR) solves before creating the PR and link it here.)

Resolves (link to issue)

# Description

Discussed internally: We originally wanted an encoding which could be implemented in other languages easier. I thought that having a manual encoding and then specifying this in some documentation would have been the best way for this. Even if this was the case, we have had two serialization bugs and the effort for folks to modify this code has become unproductive.

There are still solutions to the problem of having other languages implement a deserialization strategy, they can choose to use rmp/json for example, which is available in most languages. For now, we use rmp by default with the read and write methods. 

## Summary of changes

(Describe the changes in this PR. Point out breaking changes if any.)

## Dependency additions / changes

(If applicable.)

## Test additions / changes

(If applicable.)

# Checklist

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [ ] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [ ] I have reviewed the changes on GitHub, line by line.
- [ ] I have ensured all changes are covered in the description.

# Additional context

(If applicable.)

### CODE REVIEW

**Comment by kevaundray** [comment]
Closes #228

File: acir/src/circuit/opcodes.rs [hunk 1]
```diff
@@ -224,44 +154,3 @@ impl std::fmt::Debug for Opcode {
         std::fmt::Display::fmt(self, f)
     }
 }
-
-#[test]
-fn serialization_roundtrip() {
```
Comment (vezenovm) [inline acir/src/circuit/opcodes.rs:229]: Couldn't hurt to still have the serialization_roundtrip test? At least here for the main circuit where we still have `read` and `write`

Comment (kevaundray) [reply acir/src/circuit/opcodes.rs:229]: I left it in for the main Circuit in `acir/src/circuit/mod.rs` -- The derive(Serialize, Deserialize) should take care of it for opcodes though

Comment (vezenovm) [reply acir/src/circuit/opcodes.rs:229]: Oops I missed that I thought you deleted it. Perhaps we should copy-paste some of the directives, opcodes, etc. from the other serialization tests into the test in `acir/src/circuit/mod.rs` for completeness sake

Comment (kevaundray) [reply acir/src/circuit/opcodes.rs:229]: Yep makes sense

**Comment by vezenovm** [comment]
I think you can remove `to_u16()` and `from_u16()` in black box func file

**Comment by kevaundray** [comment]
> I think you can remove `to_u16()` and `from_u16()` in black box func file

This was there prior to the manual serialization work so we probably want to check if this is being used in other places.

**Comment by vezenovm** [comment]
> This was there prior to the manual serialization work so we probably want to check if this is being used in other places.

I see it only being used in the `consistent_index` test in `black_box_functions.rs`

**Review by TomAFrench — APPROVED** [review]

**Pull request merged**